lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -209,6 +260,9 @@ public function map(Closure $func);
      * @return Collection[] An array with two elements. The first element contains the collection
      *                      of elements where the predicate returned TRUE, the second element
      *                      contains the collection of elements where the predicate returned FALSE.
+     *
+     * @psalm-param Closure(TKey, T):bool $p
```

Comment:
```suggestion
     * @psalm-param Closure(TKey=, T=):bool $p
```
As discussed previously, sometimes it makes sense to partition collection randomly or based on keys only.